File: tests/bench/crosslang/c/u64.c
/* Порт tests/bench/programs/U64Bench.eat 1:1 — 64-битная беззнаковая
 * арифметика, деление, широкие сдвиги. Вывод обязан совпасть байт-в-байт
 * с EATLang-оригиналом при том же REPEAT. */
#include <stdint.h>
#include <stdio.h>

static const uint32_t REPEAT = 1;

static uint64_t work(uint64_t seed) {
    uint64_t acc = seed;
    for (uint32_t k = 0; k < 200u; k++) {
        for (uint32_t i = 0; i < 1000u; i++) {
            acc = (acc * 31u + (uint64_t)i) % 281474976710656ull;
            acc = acc ^ (acc >> 33);
            acc = acc + acc / (uint64_t)((i & 7u) + 2u);
        }
    }
    return acc % 65536u;
}

int main(void) {
    uint64_t acc = 7u;
    for (uint32_t r = 0; r < REPEAT; r++) {
        acc = work(acc);
    }
    uint32_t c = (uint32_t)acc;
    printf("checksum %u\n", c);
    return 0;
}

File: tests/bench/crosslang/go/u64.go
// Порт tests/bench/programs/U64Bench.eat 1:1 — 64-битная беззнаковая
// арифметика, деление, широкие сдвиги. Вывод обязан совпасть байт-в-байт
// с EATLang-оригиналом при том же REPEAT.
package main

import "fmt"

const REPEAT uint32 = 1

func work(seed uint64) uint64 {
	acc := seed
	for k := uint32(0); k < 200; k++ {
		for i := uint64(0); i < 1000; i++ {
			acc = (acc*31 + i) % 281474976710656
			acc = acc ^ (acc >> 33)
			acc = acc + acc/((i&7)+2)
		}
	}
	return acc % 65536
}

func main() {
	acc := uint64(7)
	for r := uint32(0); r < REPEAT; r++ {
		acc = work(acc)
	}
	c := uint32(acc)
	fmt.Printf("checksum %d\n", c)
}
